backend/dataall/modules/dataset_sharing/api/resolvers.py

```diff
@@ -0,0 +1,236 @@
+
+
+def get_share_object(context, source, shareUri: str = None):
+    return ShareObjectService.get_share_object(uri=shareUri)
+
+
+def resolve_user_role(context: Context, source: ShareObject, **kwargs):
+    if not source:
+        return None
+    with context.engine.scoped_session() as session:
+        dataset: Dataset = DatasetRepository.get_dataset_by_uri(session, source.datasetUri)
+        if dataset and dataset.stewards in context.groups:
+            return ShareObjectPermission.Approvers.value
+        if (
+            source.owner == context.username
+            or source.principalId in context.groups
+            or dataset.owner == context.username
+            or dataset.SamlAdminGroupName in context.groups
+        ):
+            return ShareObjectPermission.Requesters.value
+        if (
+            dataset and dataset.stewards in context.groups
+            and (
+                source.owner == context.username
+                or source.principalId in context.groups
+                or dataset.owner == context.username
+                or dataset.SamlAdminGroupName in context.groups
+            )
+        ):
+            return ShareObjectPermission.ApproversAndRequesters.value
+        else:
+            return ShareObjectPermission.NoPermission.value
+
+
+def resolve_dataset(context: Context, source: ShareObject, **kwargs):
+    if not source:
+        return None
+    with context.engine.scoped_session() as session:
+        ds: Dataset = DatasetRepository.get_dataset_by_uri(session, source.datasetUri)
+        if ds:
+            env: models.Environment = db.api.Environment.get_environment_by_uri(session, ds.environmentUri)
+            return {
+                'datasetUri': source.datasetUri,
+                'datasetName': ds.name if ds else 'NotFound',
+                'SamlAdminGroupName': ds.SamlAdminGroupName if ds else 'NotFound',
+                'environmentName': env.label if env else 'NotFound',
+                'AwsAccountId': env.AwsAccountId if env else 'NotFound',
+                'region': env.region if env else 'NotFound',
+                'exists': True if ds else False,
+            }
+
+
+def union_resolver(object, *_):
+    if isinstance(object, DatasetTable):
+        return 'DatasetTable'
+    elif isinstance(object, DatasetStorageLocation):
+        return 'DatasetStorageLocation'
+
+
+def resolve_principal(context: Context, source: ShareObject, **kwargs):
+    if not source:
+        return None
+
+    with context.engine.scoped_session() as session:
+        return get_principal(
+            session, source.principalId, source.principalType, source.principalIAMRoleName, source.environmentUri, source.groupUri
+        )
+
+
+def resolve_group(context: Context, source: ShareObject, **kwargs):
+    if not source:
+        return None
+    return source.groupUri
+
+
+def resolve_consumption_data(context: Context, source: ShareObject, **kwargs):
```

Comment:
I left those three intentionally. 
If we move everything into the services we will end up having only something like in the resolvers 
```
def submit_share_object(context: Context, source, shareUri: str = None):
    return ShareObjectService.submit_share_object(uri=shareUri)
```
That would not make much sense. 
In my opinion, the resolvers are needed for the following thing:
1) Validation of requests
2) Transform the request to the backend data as it needed
3) Transform the response to the user data. 
The 2 and 3 are not used much in the project because we use default models to pass the data, but if we want to modify a payload of the response I'd rather did it in the resolvers. This way a service method can be more generic, but responses can vary  depends on user request. For example,  we can have `DatasetService.get_dataset` and three resolvers `get_dataset_metadata`, `get_aws_dataset_params` and `get_full_dataset`. All there call the same service method, but use different method. I think these `resolve_consumption_data` and `resolve_dataset` is a good example. 
About `resolve_user_role`: I left it as well since it manipulates with the request data much by using `source`